webapp/src/main/java/com/box/l10n/mojito/service/ai/openai/OpenAILLMService.java

```diff
@@ -190,15 +264,40 @@ private AICheckResult parseResponse(
     return result;
   }
 
-  private static String getFormattedPrompt(String prompt, String sourceString, String comment) {
-    String systemPrompt = "";
+  private static String getStringChecksFormattedPrompt(
+      String prompt, String sourceString, String comment) {
+    String formattedPrompt = "";
     if (prompt != null) {
-      systemPrompt =
+      formattedPrompt =
           prompt
               .replace(SOURCE_STRING_PLACEHOLDER, sourceString)
               .replace(COMMENT_STRING_PLACEHOLDER, comment);
     }
-    return systemPrompt;
+    return formattedPrompt;
+  }
+
+  private static String getTranslationFormattedPrompt(
+      String prompt, TMTextUnit tmTextUnit, String sourceBcp47Tag, String targetBcp47Tag) {
+    String formattedPrompt = "";
+    if (prompt != null) {
+      formattedPrompt =
+          prompt
+              .replace(SOURCE_STRING_PLACEHOLDER, tmTextUnit.getContent())
+              .replace(
+                  COMMENT_STRING_PLACEHOLDER,
```

Comment:
Given that this is part of the prompt, is it safe to have a final formatted prompt looking like:

> The string to translate is: "Hello Pinterest user!" and the context for is: 

In practice, I might expect the responses to misbehave in those instances, with the LLM trying to complete the missing information where it's missing.

Ideally, the final prompt in that situation would not even include any reference to a context when it is blank, e.g.:
> The string to translate is: "Hello Pinterest user!" 

Any thoughts on this? This might require a different type of marking optional placeholders in the source prompts if we decide we need to "clean" up the prompt in those instances where we are missing information (same for Plural forms below, context string etc.).

Reply:
I don't think it's wise for us to attempt to automate the prompt clean up here, as we won't know at runtime how the prompt is structured and how much of it we should remove. The likelihood is that'll cause more issues than it solves at runtime.

The prompt itself should be capable of handling that some of these placeholders might be empty. Maybe mojito will inject a string like `N/A` when a placeholder is empty and the prompt has handling for that?

Reply:
I don't have a clear idea in my head of how the prompt could achieve that cleanly by itself.
Having that type of N/A information risk reducing the quality of the outputs as well adding confusing information to the context.

I'm not suggesting we try to optimistically "figure out" what'd we need to remove, but to provide a different templating mechanic for optional/not-guaranteed fields instead, e.g.:

> To use the COMMENT_STRING_PLACEHOLDER in your prompt, use the following format:
> Translate the following string to French: "[mojito_source_string]". {{optional: The string context for this is: "[mojito_comment_string]"}}
(I'm sure we can think of a better/easier to parse format)

Where, we match on {{optional: ..} - and remove the whole thing if we don't have any comment for example (or remove the optional: markup and insert the placeholder)

So the final prompt would either be (depending on the existence of a comment): 
> Translate the following string to French: "Hello world". 

or

> Translate the following string to French: "Hello world". The string context for this is: this a string that welcomes users into the world!